Feature/data

tnyandoro opened the pull request:
The are the anaylse queries that we performed from the initial queries and the refactored and the show the improvement of the query data(time)

For the first query we did an 
explain analyze SELECT COUNT(*) FROM visits where animal_id = 4;
![image](https://user-images.githubusercontent.com/30318155/135353616-c47c2a07-55a8-4103-9072-e65ec74a2486.png)

then used index changed it to
explain analyze SELECT COUNT(visits) FROM visits where animal_id = 4;
![image](https://user-images.githubusercontent.com/30318155/135353708-f7f04591-ec8b-4179-b02c-a789bc18ed31.png)


For the second query we did
explain analyze SELECT * FROM visits where vet_id = 2;
![image](https://user-images.githubusercontent.com/30318155/135353765-66a422d1-65c0-4b7c-9c7f-f112fc1f86eb.png)

then we used index instead of going through the whole table 
explain analyze SELECT visits FROM visits where vet_id = 2;
![image](https://user-images.githubusercontent.com/30318155/135353837-30588af8-3136-4d1c-84f8-571baea212b1.png)


For the third query we did 
explain analyze SELECT * FROM owners where email = 'owner_18327@mail.com';
![image](https://user-images.githubusercontent.com/30318155/135353910-7e31764b-2151-4780-970a-d7a8e0a92bca.png)

then indexed using the owners 
explain analyze SELECT owners FROM owners where email = 'owner_18327@mail.com';
![image](https://user-images.githubusercontent.com/30318155/135354063-7e087ba3-fb36-4391-87db-547d8c23b0a1.png)

The result for initial queries are found in the screenshot folder and the refactored queries results are found in the improved

tnyandoro committed:
Initital data commit

tnyandoro committed:
Add index the count from vists

tnyandoro committed:
update the database

tnyandoro committed:
update the readme

tnyandoro committed:
Push for review

DeliceLydia reviewed (changes requested):
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

- [x] The second query the execution time AFTER the improvements is higher than BEFORE the improvements Kindly work on that and this can be achieved by using indexes
- [x] The query planner output in the screenshot `after` is not using the index (there is no `Index scan` in the output) I suggest that you use `indexes` and take screenshots of the `explain analyze` output one more time

### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

elisha2kyakpo1 committed:
Add index for visits and owners

elisha2kyakpo1 committed:
Fix merge conflict

tnyandoro commented:
The second query the execution time AFTER the improvements is higher than BEFORE the improvements Kindly work on that and this can be achieved by using indexes, we have added the indexes and the time has improved 
![image](https://user-images.githubusercontent.com/30318155/135451575-e2faaa24-0fc7-4059-8e7d-eef77333e9b7.png)

The query planner output in the screenshot after is not using the index (there is no Index scan in the output) I suggest that you use indexes and take screenshots of the explain analyze output one more time
We have added the indexes and query planner now scans the indexes

![image](https://user-images.githubusercontent.com/30318155/135451700-28f531b8-4795-4f5a-8fd4-938617fdcdad.png)

DeliceLydia reviewed (approved):
# Congratulations🎉🎉

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
